### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by samrocksc**
Learning to do things is good!

### CODE REVIEW

File: src/checkbox.js [hunk 1]
```diff
@@ -12,16 +12,19 @@ const baseClasses = {
...
   }
 
   componentWillUnmount(){
-    const node = React.findDOMNode(this);
+    const node = this._element;
     mdl.downgradeElements(node);
   }
 
   render(){
+
```
Comment (phated) [inline src/checkbox.js]: remove this extra whitespace

File: src/progress.js [hunk 2]
```diff
@@ -1,4 +1,8 @@
 'use strict';
+// TODO needs to be able to have a width be determined
+// TODO Add Buffering option
+// TODO Option for using spinner instead
```
Comment (phated) [inline src/progress.js]: spinner is a different component, it should **not** be a prop

File: src/text-field.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 'use strict';
-
+//TODO change classname to camelcase
```
Comment (phated) [inline src/text-field.js]: ?

File: src/tooltip.js [hunk 4]
```diff
@@ -10,38 +11,31 @@ const baseClasses = {
...
-
   render(){
+    
     const {
       children,
       className,
       large
     } = this.props;
 
+    let { id } = this.props;
```
Comment (phated) [inline src/tooltip.js]: ~~don't like naming this id~~

File: src/tooltip.js [hunk 5]
```diff
@@ -10,38 +11,31 @@ const baseClasses = {
...
-    mdl.upgradeElement(node, 'MaterialTooltip');
-  }
-
-  componentWillUnmount(){
-    const node = React.findDOMNode(this);
-    mdl.downgradeElements(node);
-  }
-
   render(){
+
```
Comment (phated) [inline src/tooltip.js]: remoe this extra whitespace

File: src/tooltip.js (hunk 4 above)
Comment (phated) [inline src/tooltip.js]: ~~nevermind, it seems that is consistent with other components~~

File: src/text-field.js [hunk 6]
```diff
@@ -45,7 +45,7 @@ class Textfield extends React.Component {
     let {
       id
     } = this.props;
-
+
```
Comment (phated) [inline src/text-field.js]: you should set up your editor to remove trailing whitespace on file save

File: src/slider.js [hunk 7]
```diff
@@ -1,5 +1,8 @@
 'use strict';
-
+//TODO add Width
+//TODO min
+//TODO max
+//TODO tabindex
```
Comment (phated) [inline src/slider.js]: why are any of these things TODOs?  All of those pass through as props

File: src/progress.js [hunk 8]
```diff
@@ -1,4 +1,8 @@
 'use strict';
+// TODO needs to be able to have a width be determined
```
Comment (phated) [inline src/progress.js]: why?  If you put it in a constrained width component, it should just take up the space given

File: src/badge.js [hunk 9]
```diff
@@ -9,17 +9,17 @@ const baseClasses = {
 };
 
 class Badge extends React.Component {
-
+/*
```
Comment (phated) [inline src/badge.js]: don't submit commented code

File: src/badge.js [hunk 10]
```diff
@@ -33,9 +33,14 @@ class Badge extends React.Component {
       'mdl-badge--overlap': overlap
     }, className);
 
+//    const saveRef = (element) => this._element = element;
```
Comment (phated) [inline src/badge.js]: don't submit commented code

File: src/badge.js [hunk 11]
```diff
@@ -33,9 +33,14 @@ class Badge extends React.Component {
...
+//    <a {...this.props} ref={saveRef} className={classes}>
+//       {children}
+//      </a>
+
     return (
       <a {...this.props} className={classes}>
-        {children}
+       {children}
```
Comment (phated) [inline src/badge.js]: indentation is off

**Comment by samrocksc** [comment]
This has been updated

File: src/badge.js [hunk 12]
```diff
@@ -10,17 +10,8 @@ const baseClasses = {
...
-    mdl.upgradeElement(node, 'MaterialBadge');
-  }
-
-  componentWillUnmount(){
-    const node = React.findDOMNode(this);
-    mdl.downgradeElements(node);
-  }
-
   render(){
+
```
Comment (phated) [inline src/badge.js]: remove extra whitespace

File: src/badge.js [hunk 13]
```diff
@@ -35,7 +26,7 @@ class Badge extends React.Component {
 
     return (
       <a {...this.props} className={classes}>
-        {children}
+      {children}
```
Comment (phated) [inline src/badge.js]: indentation is still wrong.  Take a look at your diffs when you push stuff

File: src/checkbox.js (hunk 1 above)
Comment (phated) [inline src/checkbox.js]: missed this

File: src/progress.js [hunk 14]
```diff
@@ -32,8 +36,11 @@ class Progress extends React.Component {
       'mdl-progress__indeterminate': indeterminate
     }, className);
 
+    const saveRef = (element) => this._element = element;
+
+
```
Comment (phated) [inline src/progress.js]: remove extra whitespace

File: src/text-field.js [hunk 15]
```diff
@@ -20,16 +19,17 @@ const labelBaseClasses = {
...
 
   componentDidMount(){
-    const node = React.findDOMNode(this);
+    const node = this._element;
     mdl.upgradeElement(node, 'MaterialTextfield');
   }
 
   componentWillUnmount(){
-    const node = React.findDOMNode(this);
+    const node= this._element;
```
Comment (phated) [inline src/text-field.js]: space missing before =

**Comment by samrocksc** [comment]
Question: when doing return within render(), why do we add whitespace after render, but not return?

**Comment by haoxins** [comment]
@phated Maybe, it's better to add a `.jscsrc` for this repo

File: example/index.js [hunk 16]
```diff
@@ -1,11 +1,56 @@
...
+      <Spinner active/>
+
+    <h4>Switches</h4>
+      <Switch id='test' label='Switch' ripple/>
+
+    <h4>Input Fields</h4>
+      <Textfield type='text' />
+
+    <h4>Tooltips</h4>
+      <span id='tt2'>Hover Me</span>
```
Comment (phated) [inline example/index.js:49]: I don't think we should use `id`, as props are passed down to the underlying children which will end up putting 2 elements with the same `id` in the DOM

File: example/index.js [hunk 17]
```diff
@@ -1,11 +1,56 @@
...
+      <Badge data-badge="2">Badge</Badge>
+
+    <h4>Buttons</h4>
+      <Button>Test Button</Button>
+
+    <h4>Checkboxes</h4>
+      <Checkbox />
+
+    <h4>Icon Toggle</h4>
+      <IconToggle ripple id='2' label='test' />
```
Comment (phated) [inline example/index.js]: remove the `id` from this component

Comment (phated) [inline example/index.js:27]: Add another IconToggle without `id` to test generation and make the label "Without id".  Change the label of this one to "With id"

**Comment by phated** [comment]
@coderhaoxin I agree but that wouldn't solve the issue of adding unnecessary whitespace to the diff

File: src/tooltip.js (hunk 4 above)
Comment (phated) [inline src/tooltip.js]: switch this back to htmlFor

File: src/tooltip.js [hunk 18]
```diff
@@ -10,38 +10,31 @@ const baseClasses = {
...
-      </span>
+      </div>
     );
   }
 }
 
 Tooltip.propTypes = {
   className: React.PropTypes.string,
-  htmlFor: React.PropTypes.string.isRequired,
+  id: React.PropTypes.string.isRequired,
```
Comment (phated) [inline src/tooltip.js]: switch this back to htmlFor

File: src/tooltip.js [hunk 19]
```diff
@@ -10,38 +10,31 @@ const baseClasses = {
...
 
+    let { id } = this.props;
+
     const classes = classnames(baseClasses, {
       'mdl-tooltip--large': large
     }, className);
 
     return (
-      <span {...this.props} className={classes}>
+      <div htmlFor={id} className={classes}>
```
Comment (phated) [inline src/tooltip.js]: switch this back to splatting props

Comment (phated) [inline src/tooltip.js]: why was this switched from span to div

**Pull request merged**